index.js

```diff
@@ -0,0 +1,254 @@
+/**
+ * This plugin allows us to create shared chunks which contain all module
+ * dependencies that don't already exist in an existing shared chunk.
+ *
+ * All dependencies required to load a shared chunk appear in the 'parents'
+ * array on each chunk.
+ *
+ * A chunk is webpack terminology for a set of modules (.js files).  Entry
+ * chunks are special chunks which initially contain the entry point (a .js
+ * file) and all files in its dependency tree.
+ *
+ * As optimization of chunks occur, modules are moved out of entry chunks and
+ * into new chunks that this plugin creates.  The plugin also updates
+ * dependencies between chunks (via the 'parents' property on chunks) based on
+ * whether a module already exists in shared chunk or not.
+ *
+ * Shared chunks are created in the order they are specified.
+ *
+ * A set of shared chunks can be specified in a webpack.config.js like so:
+ *
+ *  plugins: [
+ *      new SharedChunkPlugin({
+ *          specs: [
+ *              {
+ *                  name: "vendor",
+ *                  selectedChunks: ["foo", "bar", "baz", "qux"],
+ *                  moduleFilter: (module) =>
+ *                      /vendor/.test(module.resource),
+ *              },
+ *              {
+ *                  name: "components",
+ *                  selectedChunks: ["foo", "bar", "baz", "qux"],
+ *                  moduleFilter: (module) =>
+ *                      /components/.test(module.resource),
+ *              },
+ *              {
+ *                  name: "feature-shared",
+ *                  selectedChunks: ["foo", "bar", "baz", "qux"],
+ *                  moduleFilter: (module) =>
+ *                      /features/.test(module.resource),
+ *                  minChunks: 2,
+ *              },
+ *          ]
+ *      })
+ *  ],
+ *
+ * Each of the 'specs' can contain the following options:
+ * - name: the name of the chunk being generated
+ * - selectedChunks: an array of entry chunks to consider when looking for
```

Comment:
It seems like this should be optional, if it's not provided then just have it default to using `allChunks`.

Reply:
Good call, this is how `CommonsChunkPlugin` treats `chunks`.  I called it `selectedChunks` here because it's a better name and that's what `CommonsChunkPlugin` uses internally.